### ISSUE TITLE
MyQ

### ISSUE THREAD

**Issue by erfinb** [nfarina/homebridge#368]
Cannot get MyQ to connect, where do we locate the Device ID?

**Comment by nfarina** [comment]
Have you tried omitting the device ID? If you only have one garage door, it should discover the ID automatically. You only need to enter the correct username and password in config.json.

**Comment by erfinb** [comment]
I tried that and it says that it cannot identify anything. 😕

**Comment by erfinb** [comment]
Would it make a difference if the MyQ is built into my garage door opener?

**Comment by nfarina** [comment]
It could, yes. Could you add a bit of logging to print out what your `device["MyQDeviceTypeName"]` value is? It's in LiftMaster.js:100

**Comment by erfinb** [comment]
How would I go about getting that to log, and by the way thank you for creating this bridge it's a blast and incredibly useful!

**Comment by nfarina** [comment]
Write something like this on line 99:

    console.log(device["MyQDeviceTypeName"]);

**Comment by nfarina** [comment]
And thanks :)

**Comment by erfinb** [comment]
Okay, so here is what I have:

This is from Terminal if I try to do anything with the garage door:

[Garage Door] Logged in with user ID 0
[Garage Door] Error: Couldn't find a door device, or the ID you specified isn't associated with your account

------------

This is what is in the config file:

"accessories": [
{
"accessory": "LiftMaster",
"name": "Garage Door",
"description": "This shim supports LiftMaster garage door openers that are already internet-connected to the 'MyQ' service.",
"requiredDeviceId": "<ID of door if you have multiple doors, prompted by shim during startup if needed>",
"username": "************",
"password" : "***********"
},

--------------
My username (email) and password are in there normally :)

If i leave the // in front of the "requiredDeviceID" then I get errors in compiling (as shown below):

--------------

There was a problem reading your config.json file.
Please try pasting your config.json file here to validate it: http://jsonlint.com

/Users/Family/homebridge/app.js:43
  throw err;
  ^

SyntaxError: Unexpected token /
    at Object.parse (native)
    at Object.<anonymous> (/Users/Family/homebridge/app.js:37:17)
    at Module._compile (module.js:425:26)
    at Object.Module._extensions..js (module.js:432:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Function.Module.runMain (module.js:457:10)
    at startup (node.js:136:18)
    at node.js:972:3
error: Forever detected script exited with code: 1

--------------

In the LiftMaster.js file under the accessories folder this is what I have placed at line 99:

// If we haven't explicity specified a door ID, we'll loop to make sure we don't have multiple openers, which is confusing
if (!that.requiredDeviceId) {
var thisDeviceId = device.MyQDeviceId;
var thisDoorName = "Unknown";
for (var j = 0; j < device.Attributes.length; j ++) {
console.log(device["MyQDeviceTypeName"]);
var thisAttributeSet = device.Attributes[j];
if (thisAttributeSet.AttributeDisplayName == "desc") {
thisDoorName = thisAttributeSet.Value;
break;
}

------

However when I go to the log folder in the homebridge directory, there isn't anything there....



On Nov 04, 2015, at 08:21 AM, Nick Farina <notifications@github.com> wrote:

Write something like this on line 99:
console.log(device["MyQDeviceTypeName"]);

**Comment by nfarina** [comment]
You can't "comment out" things in JSON files - you'll have to remove the line with your ID manually.

**Comment by erfinb** [comment]
Okay, when I remove it I still get the same errors that I posted before :)

On Nov 05, 2015, at 03:36 PM, Nick Farina <notifications@github.com> wrote:

You can't "comment out" things in JSON files - you'll have to remove the line with your ID manually.

**Comment by erfinb** [comment]
I have adjusted it to this:

        {
            "accessory": "LiftMaster",
            "name": "Garage Door",
            "description": "This shim supports LiftMaster garage door openers that are already internet-connected to the 'MyQ' service.",
            "username": “*******",
            "password": “********"
        }

and am still getting this error 🙁

[Garage Door] Logged in with user ID 0
[Garage Door] Error: Couldn't find a door device, or the ID you specified isn't associated with your account

Where would I find a device ID for this?

**Comment by nfarina** [comment]
You'll need to dig into the `LiftMaster.js` source code and write some code - specifically what I recommended above - because I think your device is getting filtered out as an unknown device type, and without having access to your account, I'll need your help identifying what "magic string" your device actually is.

**Comment by erfinb** [comment]
Is this where you wanted me to insert console.log(device["MyQDeviceTypeName"]); on line 99, where would I get the log file for you when this logs?

**Comment by erfinb** [comment]
// If we haven't explicity specified a door ID, we'll loop to make sure we don't have multiple openers, which is confusing
            if (!that.requiredDeviceId) {
              var thisDeviceId = device.MyQDeviceId;
              var thisDoorName = "Unknown";
              for (var j = 0; j < device.Attributes.length; j ++) {
                console.log(device["MyQDeviceTypeName"]);
                var thisAttributeSet = device.Attributes[j];
                if (thisAttributeSet.AttributeDisplayName == "desc") {
                  thisDoorName = thisAttributeSet.Value;
                  break;
                }
              }
              foundDoors.push(thisDeviceId + " - " + thisDoorName);
              that.deviceId = thisDeviceId;
            }

**Comment by erfinb** [comment]
My MyQ Device is a Chamberlain is that making a difference?

**Comment by erfinb** [comment]
I figured it out after a little searching on the web, found out how to access the data dump for my personal data and started sifting through it. I found out that my Garage Door is identified as a "Garage Door Opener WGDO" I edited the code and bam it works.  Below is what I altered in the LiftMaster.js file to make this work! Thank you again for your help!

          if (device["MyQDeviceTypeName"] == "Garage Door Opener WGDO" || device["MyQDeviceTypeName"] == "VGDO") {
